### ISSUE TITLE
Provide full dependency lists for building .NET Core

### ISSUE THREAD

**Issue by aaronfranke** [dotnet/corefx#40722]
Neither the [dogfooding](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/dogfooding.md) article nor the [Microsoft docs](https://docs.microsoft.com/en-us/dotnet/core/build/) article provide a full list of dependencies required to build .NET Core.

In fact, the documentation that is present is a bit outdated...

>The build currently depends on the following prerequisites:
>
>    Git
>    CMake
>    Python
>    a C++ compiler.

...since it only asks for "a C++ compiler", I have GCC installed, but it refused to build unless I had Clang installed. I also have Git, CMake, and Python installed, and it still doesn't build.

It told me it needed `libcurl4-openssl-dev` which I installed, then it needed `libkrb5-dev`, then `libssl-dev`, then it finally was able to build successfully. Rather than retrying and continually installing new libraries, it would be awesome if the docs had something like this:

    sudo apt install libcurl4-openssl-dev libkrb5-dev libssl-dev git cmake clang python3

The docs should also mention which versions are required, have install commands for multiple distros, and have download links for other platforms such as Windows and Mac.

**Comment by Gnbrkm41** [comment]
https://github.com/dotnet/corefx/wiki/Development-environment-setup-for-Linux >

**Comment by Gnbrkm41** [comment]
Also, I'm not sure if you're trying to build corefx or coreclr; it's only coreclr that requires Python IIRC.

**Comment by Gnbrkm41** [comment]
However it looks like the linked MS Docs article is outdated indeed.

For building coreclr on Linux, Please take a look at this document in coreclr repo: https://github.com/dotnet/coreclr/blob/master/Documentation/building/linux-instructions.md

**Comment by aaronfranke** [comment]
I was trying to build CoreFx. and that command I posted is the list of dependencies I know about, except perhaps Python shouldn't be in there as you said.

>FWIW, I don't think the first docs you linked isn't about building from source

There is a section called "More Advanced Scenario - Using your local CoreFx build". It should probably have links in the first sentence to articles on how one would go about building, not just "If you built corefx locally with build -allconfigurations", but more like "You can build CoreFx locally by following the instructions here. Then you can use the build locally on your machine."

**Comment by Gnbrkm41** [comment]
Yeah. I think it's better to update the docs to include the links to build instructions in corefx rather than keeping a separate copy.

**Comment by danmosemsft** [comment]
We'd welcome a PR for the doc. We plan to merge the repos which should make it easier to keep them correct.
